src/MsSqlAcceptanceTests/When_custom_finder_returns_existing_saga.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Threading.Tasks;
+using NServiceBus;
+using NServiceBus.AcceptanceTesting;
+using NServiceBus.AcceptanceTests;
+using NServiceBus.AcceptanceTests.EndpointTemplates;
+using NServiceBus.Extensibility;
+using NServiceBus.Persistence;
+using NServiceBus.Persistence.Sql;
+using NServiceBus.Sagas;
+using NUnit.Framework;
+
+[TestFixture]
+public class When_custom_finder_returns_existing_saga : NServiceBusAcceptanceTest
+{
+    [Test]
+    public async Task Should_use_existing_saga()
+    {
+        var context = await Scenario.Define<Context>()
+            .WithEndpoint<SagaEndpoint>(b => b
+                .When(session =>
+                {
+                    var startSagaMessage = new StartSagaMessage
+                    {
+                        Property = "Test"
+                    };
+                    return session.SendLocal(startSagaMessage);
+                }))
+            .Done(c => c.HandledOtherMessage)
+            .Run()
+            .ConfigureAwait(false);
+
+        Assert.True(context.FinderUsed);
+    }
+
+    public class Context : ScenarioContext
+    {
+        public bool FinderUsed { get; set; }
+        public bool HandledOtherMessage { get; set; }
+    }
+
+    public class SagaEndpoint : EndpointConfigurationBuilder
+    {
+        public SagaEndpoint()
+        {
+            EndpointSetup<DefaultServer>();
+        }
+
+        public class CustomFinder : IFindSagas<TestSaga.SagaData>.Using<SomeOtherMessage>
+        {
+            // ReSharper disable once MemberCanBePrivate.Global
+            public Context Context { get; set; }
+
+            public Task<TestSaga.SagaData> FindBy(SomeOtherMessage message, SynchronizedStorageSession session, ReadOnlyContextBag context)
+            {
+                Context.FinderUsed = true;
+
+                return session.GetSagaData<TestSaga.SagaData>(
+                    readOnlyContextBag: context,
+                    whereClause: "JSON_VALUE(Data,'$.Property') = @propertyValue",
+                    addParameters: command =>
```

Comment:
they shouldn't, IMO we should call it, maybe let them just pass key-values, or provide some custom type for this?
either way, since I can't set other elements of the command, it shouldn't be exposed there, it's a small WTF at first look